packages/validator/src/lib.js

```diff
@@ -320,30 +373,64 @@ class InvalidClaim extends Failure {
 /**
  * @implements {API.Unauthorized}
```

Comment:
I think you can have tsc enforce this by adding to the constructor last line:
```
/** @type {API.Unauthorized} */ (this);
```

Reply:
@gobengo this has came up on several of your reviews so lets make sure we're on the same page on this:

If you are under wrong impression that TS does not [enforces `@implements` directive](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#implements), your information may be out of date.

If there is another reason that I fail to recognize for doing type cast please elaborate

Reply:
You're right! Sorry for my confusion. I verified that `@implements` does seem to work with latest. I'm confused by https://github.com/microsoft/TypeScript/issues/30156#issuecomment-952667829